packages/block-editor/src/components/url-input/test/button.js

```diff
@@ -1,91 +1,161 @@
-			.simulate( 'click' );
-
-	it( 'should have a valid class name in the wrapper tag', () => {
-		const wrapper = shallow( <URLInputButton /> );
-		expect( wrapper.hasClass( 'block-editor-url-input__button' ) ).toBe(
-			true
-		);
-	} );
-	it( 'should have isPressed props set to false when url prop not defined', () => {
-		const wrapper = shallow( <URLInputButton /> );
-		expect( wrapper.find( 'ForwardRef(Button)' ).prop( 'isPressed' ) ).toBe(
-			false
-		);
-	} );
-	it( 'should have isPressed prop set to true if url prop defined', () => {
-		const wrapper = shallow( <URLInputButton url="https://example.com" /> );
-		expect( wrapper.find( 'ForwardRef(Button)' ).prop( 'isPressed' ) ).toBe(
-			true
-		);
+	it( 'should render a `Insert link` button and not be pressed when `url` is not provided', () => {
+		render( <URLInputButton /> );
+
+		expect(
+			screen.queryByRole( 'button', {
+				name: 'Edit link',
+				pressed: true,
+			} )
+		).not.toBeInTheDocument();
+
+		expect(
+			screen.getByRole( 'button', {
+				name: 'Insert link',
+				pressed: false,
+			} )
+		).toBeVisible();
 	} );
-	it( 'should have hidden form by default', () => {
-		const wrapper = shallow( <URLInputButton /> );
-		expect( wrapper.find( 'form' ) ).toHaveLength( 0 );
-		expect( wrapper.state().expanded ).toBe( false );
+
+	it( 'should render an `Edit link` button and be pressed when `url` is provided', () => {
+		render( <URLInputButton url="https://example.com" /> );
+
+		expect(
+			screen.queryByRole( 'button', {
+				name: 'Insert link',
+				pressed: false,
+			} )
+		).not.toBeInTheDocument();
+
+		expect(
+			screen.getByRole( 'button', {
+				name: 'Edit link',
+				pressed: true,
+			} )
+		).toBeVisible();
 	} );
-	it( 'should have visible form when Edit Link button clicked', () => {
-		const wrapper = shallow( <URLInputButton /> );
-		clickEditLink( wrapper );
-		expect( wrapper.find( 'form' ) ).toHaveLength( 1 );
-		expect( wrapper.state().expanded ).toBe( true );
+
+	it( 'should not render a form by default', () => {
+		render( <URLInputButton /> );
+
+		expect(
+			screen.queryByRole( 'button', { name: 'Submit' } )
+		).not.toBeInTheDocument();
 	} );
-	it( 'should call onChange function once when value changes once', () => {
-		const onChangeMock = jest.fn();
-		const wrapper = shallow( <URLInputButton onChange={ onChangeMock } /> );
-		clickEditLink( wrapper );
-		wrapper.find( URLInput ).simulate( 'change' );
-		expect( onChangeMock ).toHaveBeenCalledTimes( 1 );
+
+	it( 'should render a form when `Insert link` button is clicked', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+		render( <URLInputButton /> );
+
+		// Click the button to insert a link.
+		await user.click(
+			screen.getByRole( 'button', {
+				name: 'Insert link',
+				pressed: false,
+			} )
+		);
+
+		expect(
+			screen.getByRole( 'button', { name: 'Submit' } )
+		).toBeVisible();
```

Comment:
Nit: in the form-related tests, should we also look for (the presence or the non-presence of) a `form` element too? I believe we could straight up use the `role="form"` in the queries.

When necessary, we could then query for the submit button `within` that form

Reply:
Well, we can't use them here because that `<form />` doesn't have an accessible name, and that's a requirement for the implicit `form` role, so it doesn't have the `form` role after all. We could add an accessible name, but I wasn't sure what it would be, so decided to keep it as-is.